ios/Plugin/Plugin.swift

```diff
@@ -340,19 +340,27 @@ public class CAPHttpPlugin: CAPPlugin {
     return normalizedHeaders[header.lowercased()]
   }
   
-  func getRequestData(_ request: URLRequest, _ data: [String:Any], _ contentType: String) throws -> Data? {
+  func getRequestData(_ request: URLRequest, _ call: CAPPluginCall, _ data: [String:Any]?, _ contentType: String) throws -> Data? {
     if contentType.contains("application/json") {
-      return try setRequestDataJson(request, data)
+      return try setRequestDataJson(request, call)
     } else if contentType.contains("application/x-www-form-urlencoded") {
-      return setRequestDataFormUrlEncoded(request, data)
+      return setRequestDataFormUrlEncoded(request, data!)
     } else if contentType.contains("multipart/form-data") {
-      return setRequestDataMultipartFormData(request, data)
+      return setRequestDataMultipartFormData(request, data!)
     }
     return nil
   }
-  
-  func setRequestDataJson(_ request: URLRequest, _ data: [String:Any]) throws -> Data? {
-    let jsonData = try JSONSerialization.data(withJSONObject: data)
+
+  func setRequestDataJson(_ request: URLRequest, _ call: CAPPluginCall) throws -> Data? {
+    let jsonData: Data
+    let jsObject = call.getObject("data")
+
+    if (jsObject != nil) {
+      jsonData = try JSONSerialization.data(withJSONObject: jsObject as Any)
+    } else {
+      let jsArray = call.getArray("data", (Any).self)
```

Comment:
Without a default value here, apps will crash when trying to `POST` with `null` in the body
```suggestion
      let jsArray = call.getArray("data", (Any).self, [])
```

Reply:
Thank you for suggestion, it is committed.